File: exercises/majorityelement.rb
def majority_element(nums)
  answer = 0
  goal = (nums.length / 2) + 1
  obj = {}
  nums.each do |num|
    obj[num] = (obj[num] || 0) + 1
    if obj[num] == goal
      answer = num
    end
  end
  answer
end

File: exercises/majorityelement.js
var majorityElement = function(nums) {
    let goal = Math.floor(nums.length / 2) + 1
    let obj = {}
    for(let i = 0; i < nums.length; i++){
      let key = nums[i]
      obj[key] = (obj[key] || 0) + 1
      if(obj[key] === goal) return key
    }
  };